classes/patterns/class-block-pattern.php

```diff
@@ -27,23 +27,37 @@ abstract public static function get_name(): string;
 	abstract public static function get_config( $params = [] ): array;
 
 	/**
-	 * Pattern constructor.
+	 * Returns the pattern classname used for tracking patterns.
 	 */
-	public static function register_all() {
-		if ( ! function_exists( 'register_block_pattern' ) ) {
-			return;
-		}
+	public static function get_classname(): string {
+		return 'is_pattern-' . preg_replace( '#[^_a-zA-Z0-9-]#', '-', static::get_name() );
```

Comment:
Linked to this [comment](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/871#discussion_r917881514)